### ISSUE TITLE
Switch to sync dist-upgrader-all

### ISSUE THREAD

**Issue by cdggcali** [aptly-dev/aptly#553]
As discussed on the [mailing list](https://groups.google.com/forum/#!topic/aptly-discuss/0g1ZbY2G5LI), in order to be able to run "do-release-upgrade" you need the dist-upgrader-all folder but currently aptly doesn't have an option to sync this directory.

**Comment by desrod** [comment]
This still hasn't been resolved, why was this issue closed after almost precisely 5 years of no activity?

**Comment by randombenj** [comment]
We closed some old issues with no activity, happy to reopen this one

**Comment by desrod** [comment]
@randombenj This is timely, as Ubuntu 22.04 is being released imminently (this week'ish) and we have lots of users and customers who are trying to use `do-release-upgrade` to upgrade their existing environments from 18.04 to 20.04/22.04, and can't do this if their software is controlled in an internal aptly repository. 

I've managed to work around the other missing piece with the default of `-with-installer` option, which brings in the missing PXE image files, including the `ubuntu-minimal` needed to help bootstrap the safe rollback of `do-release-ugprade`, but without the ability to mirror those trees into the existing aptly repository, we're leaving everyone out there inventing/hacking their own wheel to try to get this working, including some _very_ unsupportable workflows. 

Please let me know what details/information you need to help get this lifted in priority so we can help our growing users and customers who use these repository tools. 

Thanks!

**Comment by randombenj** [comment]
@desrod I do not quite get what is not working. Isn't it the intended behaviour to not mirror installers when `-with-installer` is not set? And by using `-with-installer` are the installers still missing?

**Comment by desrod** [comment]
@randombenj There are two blockers preventing `do-release-upgrade` from working successfully with an aptly-mirrored Ubuntu repositiory: 

1. `do-release-upgrade` requires pulling the `ubuntu-minimal` package among others, and by default, that does not seem to be mirrored by aptly. Adding `-with-installer` seems to pull in some missing installer packages (will be testing this further this week to confirm this addresses one part of this upgrade), and 
2. The `dist-upgrader-all` [hierarchy](http://archive.ubuntu.com/ubuntu/dists/bionic/main/dist-upgrader-all/) in the upstream repository is not mirrored at all, even when including `-with-installer` (not sure it should, but could be part of the same "intention"). 

Any customer/person attempting to do a `do-release-upgrade` from 16.04, 18.04, 20.04 to a newer release, will fail, because these components are not available in the internal, mirrored repository. For customers/persons who have _zero_ Internet access to their datacenters, restrictive proxy policies, this presents a difficult challenge. 

At the moment, upgrading Ubuntu releases using an aptly repository is blocked, until we can find supported workarounds/solutions to the above 2 issues. 

Thoughts?

**Comment by randombenj** [comment]
Indeed, it would be good if `do-release-upgrade` would be supported with aptly-mirrored repositories. Do you have an example list of files/folder which are currently not mirrored correctly?

**Comment by desrod** [comment]
@randombenj So far, there are several directories within `http://archive.ubuntu.com/ubuntu/dists/bionic/main/` which are not being pulled in, including an important one `dist-upgrader-all` which is used by `do-release-upgrade` to facilitate the upgrade. 

I'll have a much more detailed breakdown today, as I'm pulling these pieces apart behind a restrictive environment with no direct Internet access and several different proxy configurations. 

Stay tuned...
